interfaces/builtin/custom_device.go

```diff
@@ -403,6 +405,29 @@ func (iface *customDeviceInterface) UDevConnectedPlug(spec *udev.Specification,
 		deviceRules[deviceName] = rule.String()
 	}
 
+	// Generate a basic udev rule for each device which was not explicitly
+	// matched by a kernel device name in the "udev-tagging" attribute.
+	// Since each kernel device name matches exactly one device in
+	// allDevicePaths (see: validateUDevTaggingRule), then any device which is
+	// not matched by any kernel device should get a basic KERNEL="<device>"
+	// rule.
+	for _, devicePath := range allDevicePaths {
+		if !strings.HasPrefix(devicePath, "/dev/") {
+			continue
+		}
+		foundMatch := false
+		for deviceName := range deviceRules {
+			if strings.HasSuffix(devicePath, deviceName) {
+				foundMatch = true
+				break
+			}
+		}
+		if !foundMatch {
+			deviceName := devicePath[5:]
```

Comment:
I know this is pre-existing code but I feel like we should make it more clearer now that this is moved, the `devicePath[5:]` feels quite obscure to me, especially now that the connection to `if !strings.HasPrefix(devicePath, "/dev/") {` is more spread out. Maybe just:
```go
-                       deviceName := devicePath[5:]
+                       deviceName := devicePath[len("/dev/"):]
```
I know it's less efficient but I prefer the readability here

Reply:
I just addressed it myself so that this PR can be merged :)